fastscanner/adapters/fundamental/eodhd.py

```diff
@@ -1,5 +1,176 @@
+import json
+import logging
+import os
+from dataclasses import asdict
+from typing import Dict
+
+import httpx
+import pandas as pd
+
+from fastscanner.pkg import config
 from fastscanner.services.indicators.ports import FundamentalData
 
+logger = logging.getLogger(__name__)
+
 
 class EODHDFundamentalStore:
-    def get(self, symbol: str) -> "FundamentalData": ...
+    CACHE_DIR = os.path.join("data", "fundamentals")
+
+    def __init__(self, base_url: str, api_key: str):
+        self._base_url = base_url
+        self._api_key = api_key
+
+    def get(self, symbol: str) -> FundamentalData:
+        logger.info(f"Retrieving fundamentals for: {symbol}")
+
+        cached = self._load_cached(symbol)
+        if cached:
+            logger.info(f"Loaded {symbol} fundamentals from cache.")
+            return cached
+
+        fundamentals = self._fetch_fundamentals(symbol)
+        market_cap = self._fetch_market_cap(symbol)
+
+        fd = self._parse_data(fundamentals, market_cap)
+        try:
+            self._store(symbol, fd)
+            logger.info(f"Stored fundamental data for {symbol}")
+        except Exception as e:
+            logger.error(f"Failed to process {symbol}: {e}")
+            raise
+        return fd
+
+    def _fetch_fundamentals(self, symbol: str) -> Dict:
+        filters = "General::Code,General,Earnings,SharesStats"
+        url = f"{self._base_url}/fundamentals/{symbol}?filter={filters}&api_token={self._api_key}&fmt=json"
```

Comment:
Let's pass the params argument to the request instead of building the URL ourselves.